websites/A/Animes Portal/presence.ts

```diff
@@ -0,0 +1,127 @@
+const presence = new Presence({
+  clientId: '924791712944099389'
+});
+
+const filterCodes: any = {
+  366: 'bgsub'
+}
+
+presence.on('UpdateData', async () => {
+
+  const presenceData: PresenceData = {
+    largeImageKey: 'logo',
+  };
+
+  if (document.location.pathname === '/') {
+    assign(presenceData, {
+      details: 'Home',
+      state: 'Viewing the home page'
+    });
+  } else if (document.location.pathname.startsWith('/animes')) {
+    let url = new URL(document.location.href);
+    let filter: any = url.searchParams.get('filters');
+    let paths = document.location.pathname.split('/');
+    if (!paths[0]) paths.shift();
+    let filters: Array<string> = [];
+
+    if (filter) {
+      filter.split(',').map((x: string) => parseInt(x)).forEach((f: number) => {
+        if (filterCodes[f]) filters.push(filterCodes[f])
+      })
+    }
+
+    if (paths[1] === 'search' && paths[2]) {
+      assign(presenceData, {
+        details: `Searching for ${paths[2].replaceAll('-', ' ')} ${filters.includes('bgsub') ? `with BG sub` : ''}`
+      });
+    } else if (paths[1]) {
+      assign(presenceData, {
+        details: `Viewing animes that starts with the keyboard ${paths[1].toUpperCase()}`
+      });
+    } else
+      assign(presenceData, {
+        details: `Viewing animes ${filters.includes('bgsub') ? `with BG sub` : ''}`
+      });
+  } else if (document.location.pathname.startsWith('/anime/')) {
+    let paths = document.location.pathname.split('/');
+    if (!paths[0]) paths.shift();
+    let uid = paths[1];
+    let eid = paths[2];
+
+    if (uid && !eid) {
+      let name = document.querySelector('body > main.animated > div.wrapper > article.rowView > header.rowView-head > h1.heading').innerHTML;
+      assign(presenceData, {
+        details: `Viewing ${name}`
+      });
+    } else if (uid && eid) {
+      let ep = document.querySelector('body > main.animated > div.wrapper > section.holder > h1.heading > b#num').innerHTML;
```

Comment:
I will make one function for this, and call it here and in line 93, but in the next update